sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,6 +877,58 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(expr: FromUTCTimestamp,
+                               override val conf: RapidsConf,
+                               override val parent: Option[RapidsMeta[_, _, _]],
+                               rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  // Adding other time zones needs to carefully consider daylight saving time adjustment.
+  private val supportedUTCTimeZones = Set("UTC", "GMT")
+
+  override def tagExprForGpu(): Unit = {
+    val timezone = GpuOverrides.extractStringLit(expr.right).getOrElse("")
+    if (timezone == null || !supportedUTCTimeZones.contains(timezone.toUpperCase)) {
+      willNotWorkOnGpu("only time zones equivalent to UTC are supported")
+    }
+  }
+
+  override def convertToGpu(timestamp: Expression, timezone: Expression): GpuExpression =
+    GpuFromUTCTimestamp(timestamp, timezone)
+}
+
+case class GpuFromUTCTimestamp(timestamp: Expression,
+                               timezone: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def left: Expression = timestamp
+  override def right: Expression = timezone
+  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, StringType)
+  override def dataType: DataType = TimestampType
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    throw new IllegalStateException("Really should not be here: " +
```

Comment:
Nit: "Really should not be here" is redundant, because that's what `IllegalStateException` already implies.